### ISSUE TITLE
Is it possible to train the model without ground truth masks?

### ISSUE THREAD

**Issue by arnabkuw** [IDT-ITI/MMFusion-IML#19]
I currently have a dataset of 100 real vs 100 fake images, but I do not have the groundtruths. Generating groundtruth for my dataset is quite difficult since the datasets are not aligned. I was wondering if it's possible to just run phase 2 training (detection) for this model, taking a previous localization checkpoint?

It seems the criterion for `ec_phase2.py` i.e. `TruForLossPhase2` still required the `gt_masks` to calculate detection loss. 

```bash
class TruForLossPhase2(torch.nn.Module):
    def __init__(self, lambda_det: float = 0.5, ignore_index: int = -1):
        super().__init__()
        self.lambda_det = lambda_det
        self.ignore_index = ignore_index
        self.criterion_detect = torch.nn.BCEWithLogitsLoss()
        self.criterion_conf = torch.nn.MSELoss(reduction='none')

    def forward(self, anomaly: Tensor, gt_mask: Tensor, conf: Tensor, detect: Tensor, label: Tensor) -> Tensor:
        anomaly = torch.softmax(anomaly, dim=1)[:, 1, :, :]
        t = gt_mask * anomaly + (1 - gt_mask) * (1 - anomaly)

        valid = gt_mask != self.ignore_index
        mse = self.criterion_conf(conf.squeeze(1), t)
        Lconf = mse[valid].mean()

        Ldet = self.criterion_detect(detect.squeeze(1), label.to(torch.float32))

        loss = Lconf + Ldet * self.lambda_det

        return loss
```

Let me know your thoughts. Any help is appreciated.
